src/simpleGit.ts

```diff
@@ -146,33 +147,43 @@ export class SimpleGit extends GitManager {
         this.plugin.setState(PluginState.pull);
         if (this.plugin.settings.updateSubmodules)
             await this.git.subModule(["update", "--remote", "--merge", "--recursive"], (err: any) => this.onError(err));
-        let lastRemoteCommitBefore: string;
-        if (!this.plugin.settings.mergeOnPull) {
-            lastRemoteCommitBefore = await this.getNewestRemoteCommit();
 
-        }
-        const pullResult = await this.git.pull([this.plugin.settings.mergeOnPull ? '--no-rebase' : '--rebase'],
```

Comment:
Previously, we were using `git pull` to fetch and merge changes.  `git pull` is a shorthand for `git fetch` followed by either `git rebase FETCH_HEAD` or `git merge FETCH_HEAD` (https://git-scm.com/docs/git-pull/2.1.4#_description).

In the new changes on the right, you'll see that we're intentionally `git fetch`-ing separately from `git merge` or `git rebase`, but otherwise these are largely functionally the same.